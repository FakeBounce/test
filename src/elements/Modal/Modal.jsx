import React, { Component } from "react";
import { Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";
import styles from "./Modal.module.scss";

class ModalElement extends Component {
  render() {
    const {
      isOpen,
      toggle,
      className,
      children,
      title,
      ModalFooterContent = null
    } = this.props;
    return (
      <div>
        <Modal
          isOpen={isOpen}
          toggle={toggle}
          className={`${styles.ModalContent} ${className}`}
        >
          <ModalHeader className={styles.ModalHeader} toggle={toggle}>
            {title}
          </ModalHeader>
          <ModalBody className={styles.ModalBody}>{children}</ModalBody>
          <ModalFooter>
            {ModalFooterContent && <ModalFooterContent />}
          </ModalFooter>
        </Modal>
      </div>
    );
  }
}

export default ModalElement;
