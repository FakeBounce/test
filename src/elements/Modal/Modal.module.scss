.ModalContent {
  > div {
    border-radius: 2px;
    padding: 30px;
    .ModalHeader {
      border-bottom: 0;
      > h5 {
        color: #1F2B3D;
        font-size: 16px;
        letter-spacing: 1.17px;
        text-transform: uppercase;
      }
      > button {
        position: absolute;
        top: 15px;
        right: 15px;
      }
    }
  }
  &:global(.Red) {
    .ModalHeader {
      > h5 {
        color: #E41818;
      }
    }
  }
}

div {
  &:global(.FormModal) {
    div {
      &:global(.modal-body) {
        padding: 0;
        overflow: hidden;
        margin: 15px;
        > div {
          margin-bottom: 15px;
          input, select {
            &::placeholder {
              color: #9E9E9E;
            }
            border: 0;
            font-size: 14px;
            color: #9E9E9E;
            border-radius: 2px;
            box-shadow: 0 2px 5px 0 rgba(0,0,0,0.24);
          }
        }
      }
      &:global(.modal-footer) {
        border-top: 0;
        button {
          border-radius: 2px;
          border-color: #5696FF;
          font-size: 14px;
          letter-spacing: 1px;
          color: #FFFFFF;
          background-color: #5696FF;
          text-transform: uppercase;
          &:global(.btn.btn-primary) {
            &:focus, &:active, &:hover {
              border-color: #5696FF;
              background-color: #5696FF;
              box-shadow: none;
              color: #FFFFFF;
            }
          }
        }
      }
    }
  }
}

div {
  &:global(.ConfirmModal) {
    div {
      &:global(.modal-content) {
        padding-bottom: 0;
        padding-right: 0;
        div {
          &:global(.modal-body) {
            div {
              color: #9E9E9E;
              font-size: 14px;
              line-height: 17px;
            }
          }
          &:global(.modal-footer) {
            border-top: 0;
            button {
              border-radius: 2px;
              font-size: 14px;
              letter-spacing: 1px;
              text-transform: uppercase;
              &:global(.btn.btn-primary) {
                border-color: #5696FF;
                color: #fff;
                background-color: #5696FF;
                &:focus, &:active, &:hover {
                  border-color: #5696FF;
                  color: #fff;
                  background-color: #5696FF;
                  box-shadow: none;
                }
              }
              &:global(.btn.btn-secondary) {
                color: #4F5561;
                background-color: #fff;
                border-color: #4F5561;
                &:focus, &:active, &:hover {
                  color: #4F5561;
                  background-color: #fff;
                  border-color: #4F5561;
                  box-shadow: none;
                }
              }
              &:global(.btn.btn-danger) {
                color: #fff;
                background-color: #E41818;
                border-color: #E41818;
                &:focus, &:active, &:hover {
                  color: #fff;
                  background-color: #E41818;
                  border-color: #E41818;
                  box-shadow: none;
                }
              }
            }
          }
        }
      }
    }
  }
}

div {
  &:global(.InformationModal) {
    div {
      &:global(.modal-content) {
        div {
          &:global(.modal-body) {
            div {
              display: flex;
              justify-content: center;
              img {
                width: 27px;
                height: 20px;
              }
            }
          }
        }
      }
      &:global(.modal-footer) {
        border-top: 0;
      }
    }
  }
}

div {
  &:global(.NewFolderModal) {
    div {
      &:global(.modal-content) {
        padding-bottom: 0;
        padding-right: 0;
        div {
          &:global(.modal-body) {
            padding: 0;
            overflow: hidden;
            margin: 15px;
            > div {
              margin-bottom: 15px;
              input {
                &::placeholder {
                  color: #9E9E9E;
                }
                border: 0;
                font-size: 14px;
                color: #9E9E9E;
                border-radius: 2px;
                box-shadow: 0 2px 5px 0 rgba(0,0,0,0.24);
              }
            }
          }
        }
      }
      &:global(.modal-footer) {
        border-top: 0;
        button {
          border-radius: 2px;
          font-size: 14px;
          letter-spacing: 1px;
          text-transform: uppercase;
          &:global(.btn.btn-primary) {
            border-color: #5696FF;
            color: #fff;
            background-color: #5696FF;
            margin-right: 5px;
            &:focus, &:active, &:hover {
              border-color: #5696FF;
              color: #fff;
              background-color: #5696FF;
              box-shadow: none;
            }
          }
          &:global(.btn.btn-secondary) {
            color: #4F5561;
            background-color: #fff;
            border-color: #4F5561;
            &:focus, &:active, &:hover {
              color: #4F5561;
              background-color: #fff;
              border-color: #4F5561;
              box-shadow: none;
            }
          }
        }
      }
    }
  }
}

div {
  &:global(.NewFileModal) {
    div {
      &:global(.modal-content) {
        padding-bottom: 0;
        padding-right: 0;
        div {
          &:global(.modal-body) {
            padding: 0;
            overflow: hidden;
            margin: 15px;
            padding-right: 20px;
            > div {
              margin-bottom: 15px;
              &:global(.dropzone) {
                padding: 10px;
                margin-right: 20px;
                width: 100%;
                min-height: 200px;
                border: 2px dashed #9E9E9E;
                border-radius: 10px;
                display: flex;
                align-items: center;
                justify-content: center;
                flex-direction: column;
                overflow: hidden;
                > div {
                  img {
                    display: none;
                  }
                }
                p {
                  text-align: center;
                  width: 350px;
                  color: #9E9E9E;
                  margin-bottom: 0;
                }
              }
              input {
                &::placeholder {
                  color: #9E9E9E;
                }
                border: 0;
                font-size: 14px;
                color: #9E9E9E;
                border-radius: 2px;
                box-shadow: 0 2px 5px 0 rgba(0,0,0,0.24);
              }
            }
          }
        }
      }
      &:global(.modal-footer) {
        border-top: 0;
        button {
          border-radius: 2px;
          font-size: 14px;
          letter-spacing: 1px;
          text-transform: uppercase;
          &:global(.btn.btn-primary) {
            border-color: #5696FF;
            color: #fff;
            background-color: #5696FF;
            margin-right: 5px;
            &:focus, &:active, &:hover {
              border-color: #5696FF;
              color: #fff;
              background-color: #5696FF;
              box-shadow: none;
            }
          }
          &:global(.btn.btn-secondary) {
            color: #4F5561;
            background-color: #fff;
            border-color: #4F5561;
            &:focus, &:active, &:hover {
              color: #4F5561;
              background-color: #fff;
              border-color: #4F5561;
              box-shadow: none;
            }
          }
        }
      }
    }
  }
}


