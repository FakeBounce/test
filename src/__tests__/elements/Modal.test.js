import React from "react";
import { shallow, configure } from "enzyme/build";
import Adapter from "enzyme-adapter-react-16/build";
import Modal from "elements/Modal/Modal";

configure({ adapter: new Adapter() });

const defaultTestingProps = {
  isOpen: false,
  title: "Title",
  toggle: () => jest.fn(),
  file: null
};

describe("Testing Modal component", () => {
  it("renders as expected without params", () => {
    const wrapper = shallow(<Modal />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders as expected with params (closed)", () => {
    const wrapper = shallow(<Modal {...defaultTestingProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders as expected with params (opened)", () => {
    const wrapper = shallow(<Modal {...defaultTestingProps} isOpen />);
    expect(wrapper).toMatchSnapshot();
  });
});

