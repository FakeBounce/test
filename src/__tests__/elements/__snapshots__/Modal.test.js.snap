// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Modal component renders as expected with params (closed) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalElement
    file={null}
    isOpen={false}
    title="Title"
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Modal
        autoFocus={true}
        backdrop={true}
        backdropTransition={
                Object {
                        "mountOnEnter": true,
                        "timeout": 150,
                      }
        }
        centered={false}
        className="undefined undefined"
        fade={true}
        isOpen={false}
        keyboard={true}
        modalTransition={
                Object {
                        "timeout": 300,
                      }
        }
        onClosed={[Function]}
        onOpened={[Function]}
        returnFocusAfterClose={true}
        role="dialog"
        scrollable={false}
        toggle={[Function]}
        unmountOnClose={true}
        zIndex={1050}
>
        <ModalHeader
                charCode={215}
                className={undefined}
                closeAriaLabel="Close"
                tag="h5"
                toggle={[Function]}
                wrapTag="div"
        >
                Title
        </ModalHeader>
        <ModalBody
                className={undefined}
                tag="div"
        />
        <ModalFooter
                tag="div"
        />
</Modal>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": true,
        "backdrop": true,
        "backdropTransition": Object {
          "mountOnEnter": true,
          "timeout": 150,
        },
        "centered": false,
        "children": Array [
          <ModalHeader
            charCode={215}
            className={undefined}
            closeAriaLabel="Close"
            tag="h5"
            toggle={[Function]}
            wrapTag="div"
>
            Title
</ModalHeader>,
          <ModalBody
            className={undefined}
            tag="div"
/>,
          <ModalFooter
            tag="div"
/>,
        ],
        "className": "undefined undefined",
        "fade": true,
        "isOpen": false,
        "keyboard": true,
        "modalTransition": Object {
          "timeout": 300,
        },
        "onClosed": [Function],
        "onOpened": [Function],
        "returnFocusAfterClose": true,
        "role": "dialog",
        "scrollable": false,
        "toggle": [Function],
        "unmountOnClose": true,
        "zIndex": 1050,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "charCode": 215,
            "children": "Title",
            "className": undefined,
            "closeAriaLabel": "Close",
            "tag": "h5",
            "toggle": [Function],
            "wrapTag": "div",
          },
          "ref": null,
          "rendered": "Title",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
            "className": undefined,
            "tag": "div",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": null,
            "tag": "div",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
                    Object {
                              "mountOnEnter": true,
                              "timeout": 150,
                            }
          }
          centered={false}
          className="undefined undefined"
          fade={true}
          isOpen={false}
          keyboard={true}
          modalTransition={
                    Object {
                              "timeout": 300,
                            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          returnFocusAfterClose={true}
          role="dialog"
          scrollable={false}
          toggle={[Function]}
          unmountOnClose={true}
          zIndex={1050}
>
          <ModalHeader
                    charCode={215}
                    className={undefined}
                    closeAriaLabel="Close"
                    tag="h5"
                    toggle={[Function]}
                    wrapTag="div"
          >
                    Title
          </ModalHeader>
          <ModalBody
                    className={undefined}
                    tag="div"
          />
          <ModalFooter
                    tag="div"
          />
</Modal>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "backdrop": true,
          "backdropTransition": Object {
            "mountOnEnter": true,
            "timeout": 150,
          },
          "centered": false,
          "children": Array [
            <ModalHeader
              charCode={215}
              className={undefined}
              closeAriaLabel="Close"
              tag="h5"
              toggle={[Function]}
              wrapTag="div"
>
              Title
</ModalHeader>,
            <ModalBody
              className={undefined}
              tag="div"
/>,
            <ModalFooter
              tag="div"
/>,
          ],
          "className": "undefined undefined",
          "fade": true,
          "isOpen": false,
          "keyboard": true,
          "modalTransition": Object {
            "timeout": 300,
          },
          "onClosed": [Function],
          "onOpened": [Function],
          "returnFocusAfterClose": true,
          "role": "dialog",
          "scrollable": false,
          "toggle": [Function],
          "unmountOnClose": true,
          "zIndex": 1050,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "charCode": 215,
              "children": "Title",
              "className": undefined,
              "closeAriaLabel": "Close",
              "tag": "h5",
              "toggle": [Function],
              "wrapTag": "div",
            },
            "ref": null,
            "rendered": "Title",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
              "className": undefined,
              "tag": "div",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": null,
              "tag": "div",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing Modal component renders as expected with params (opened) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalElement
    file={null}
    isOpen={true}
    title="Title"
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Modal
        autoFocus={true}
        backdrop={true}
        backdropTransition={
                Object {
                        "mountOnEnter": true,
                        "timeout": 150,
                      }
        }
        centered={false}
        className="undefined undefined"
        fade={true}
        isOpen={true}
        keyboard={true}
        modalTransition={
                Object {
                        "timeout": 300,
                      }
        }
        onClosed={[Function]}
        onOpened={[Function]}
        returnFocusAfterClose={true}
        role="dialog"
        scrollable={false}
        toggle={[Function]}
        unmountOnClose={true}
        zIndex={1050}
>
        <ModalHeader
                charCode={215}
                className={undefined}
                closeAriaLabel="Close"
                tag="h5"
                toggle={[Function]}
                wrapTag="div"
        >
                Title
        </ModalHeader>
        <ModalBody
                className={undefined}
                tag="div"
        />
        <ModalFooter
                tag="div"
        />
</Modal>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": true,
        "backdrop": true,
        "backdropTransition": Object {
          "mountOnEnter": true,
          "timeout": 150,
        },
        "centered": false,
        "children": Array [
          <ModalHeader
            charCode={215}
            className={undefined}
            closeAriaLabel="Close"
            tag="h5"
            toggle={[Function]}
            wrapTag="div"
>
            Title
</ModalHeader>,
          <ModalBody
            className={undefined}
            tag="div"
/>,
          <ModalFooter
            tag="div"
/>,
        ],
        "className": "undefined undefined",
        "fade": true,
        "isOpen": true,
        "keyboard": true,
        "modalTransition": Object {
          "timeout": 300,
        },
        "onClosed": [Function],
        "onOpened": [Function],
        "returnFocusAfterClose": true,
        "role": "dialog",
        "scrollable": false,
        "toggle": [Function],
        "unmountOnClose": true,
        "zIndex": 1050,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "charCode": 215,
            "children": "Title",
            "className": undefined,
            "closeAriaLabel": "Close",
            "tag": "h5",
            "toggle": [Function],
            "wrapTag": "div",
          },
          "ref": null,
          "rendered": "Title",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
            "className": undefined,
            "tag": "div",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": null,
            "tag": "div",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
                    Object {
                              "mountOnEnter": true,
                              "timeout": 150,
                            }
          }
          centered={false}
          className="undefined undefined"
          fade={true}
          isOpen={true}
          keyboard={true}
          modalTransition={
                    Object {
                              "timeout": 300,
                            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          returnFocusAfterClose={true}
          role="dialog"
          scrollable={false}
          toggle={[Function]}
          unmountOnClose={true}
          zIndex={1050}
>
          <ModalHeader
                    charCode={215}
                    className={undefined}
                    closeAriaLabel="Close"
                    tag="h5"
                    toggle={[Function]}
                    wrapTag="div"
          >
                    Title
          </ModalHeader>
          <ModalBody
                    className={undefined}
                    tag="div"
          />
          <ModalFooter
                    tag="div"
          />
</Modal>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "backdrop": true,
          "backdropTransition": Object {
            "mountOnEnter": true,
            "timeout": 150,
          },
          "centered": false,
          "children": Array [
            <ModalHeader
              charCode={215}
              className={undefined}
              closeAriaLabel="Close"
              tag="h5"
              toggle={[Function]}
              wrapTag="div"
>
              Title
</ModalHeader>,
            <ModalBody
              className={undefined}
              tag="div"
/>,
            <ModalFooter
              tag="div"
/>,
          ],
          "className": "undefined undefined",
          "fade": true,
          "isOpen": true,
          "keyboard": true,
          "modalTransition": Object {
            "timeout": 300,
          },
          "onClosed": [Function],
          "onOpened": [Function],
          "returnFocusAfterClose": true,
          "role": "dialog",
          "scrollable": false,
          "toggle": [Function],
          "unmountOnClose": true,
          "zIndex": 1050,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "charCode": 215,
              "children": "Title",
              "className": undefined,
              "closeAriaLabel": "Close",
              "tag": "h5",
              "toggle": [Function],
              "wrapTag": "div",
            },
            "ref": null,
            "rendered": "Title",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
              "className": undefined,
              "tag": "div",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": null,
              "tag": "div",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing Modal component renders as expected without params 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalElement />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Modal
        autoFocus={true}
        backdrop={true}
        backdropTransition={
                Object {
                        "mountOnEnter": true,
                        "timeout": 150,
                      }
        }
        centered={false}
        className="undefined undefined"
        fade={true}
        isOpen={false}
        keyboard={true}
        modalTransition={
                Object {
                        "timeout": 300,
                      }
        }
        onClosed={[Function]}
        onOpened={[Function]}
        returnFocusAfterClose={true}
        role="dialog"
        scrollable={false}
        toggle={undefined}
        unmountOnClose={true}
        zIndex={1050}
>
        <ModalHeader
                charCode={215}
                className={undefined}
                closeAriaLabel="Close"
                tag="h5"
                toggle={undefined}
                wrapTag="div"
        />
        <ModalBody
                className={undefined}
                tag="div"
        />
        <ModalFooter
                tag="div"
        />
</Modal>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": true,
        "backdrop": true,
        "backdropTransition": Object {
          "mountOnEnter": true,
          "timeout": 150,
        },
        "centered": false,
        "children": Array [
          <ModalHeader
            charCode={215}
            className={undefined}
            closeAriaLabel="Close"
            tag="h5"
            toggle={undefined}
            wrapTag="div"
/>,
          <ModalBody
            className={undefined}
            tag="div"
/>,
          <ModalFooter
            tag="div"
/>,
        ],
        "className": "undefined undefined",
        "fade": true,
        "isOpen": false,
        "keyboard": true,
        "modalTransition": Object {
          "timeout": 300,
        },
        "onClosed": [Function],
        "onOpened": [Function],
        "returnFocusAfterClose": true,
        "role": "dialog",
        "scrollable": false,
        "toggle": undefined,
        "unmountOnClose": true,
        "zIndex": 1050,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "charCode": 215,
            "children": undefined,
            "className": undefined,
            "closeAriaLabel": "Close",
            "tag": "h5",
            "toggle": undefined,
            "wrapTag": "div",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
            "className": undefined,
            "tag": "div",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": null,
            "tag": "div",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
                    Object {
                              "mountOnEnter": true,
                              "timeout": 150,
                            }
          }
          centered={false}
          className="undefined undefined"
          fade={true}
          isOpen={false}
          keyboard={true}
          modalTransition={
                    Object {
                              "timeout": 300,
                            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          returnFocusAfterClose={true}
          role="dialog"
          scrollable={false}
          toggle={undefined}
          unmountOnClose={true}
          zIndex={1050}
>
          <ModalHeader
                    charCode={215}
                    className={undefined}
                    closeAriaLabel="Close"
                    tag="h5"
                    toggle={undefined}
                    wrapTag="div"
          />
          <ModalBody
                    className={undefined}
                    tag="div"
          />
          <ModalFooter
                    tag="div"
          />
</Modal>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "backdrop": true,
          "backdropTransition": Object {
            "mountOnEnter": true,
            "timeout": 150,
          },
          "centered": false,
          "children": Array [
            <ModalHeader
              charCode={215}
              className={undefined}
              closeAriaLabel="Close"
              tag="h5"
              toggle={undefined}
              wrapTag="div"
/>,
            <ModalBody
              className={undefined}
              tag="div"
/>,
            <ModalFooter
              tag="div"
/>,
          ],
          "className": "undefined undefined",
          "fade": true,
          "isOpen": false,
          "keyboard": true,
          "modalTransition": Object {
            "timeout": 300,
          },
          "onClosed": [Function],
          "onOpened": [Function],
          "returnFocusAfterClose": true,
          "role": "dialog",
          "scrollable": false,
          "toggle": undefined,
          "unmountOnClose": true,
          "zIndex": 1050,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "charCode": 215,
              "children": undefined,
              "className": undefined,
              "closeAriaLabel": "Close",
              "tag": "h5",
              "toggle": undefined,
              "wrapTag": "div",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
              "className": undefined,
              "tag": "div",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": null,
              "tag": "div",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;
